src/WebService/Http/RequestFactory.php

```diff
@@ -9,18 +9,37 @@
  */
 class RequestFactory
 {
+    /**
+     * Keep the cURL resource here, so that if there are multiple API requests
+     * done the connection is kept alive, SSL resumption can be used
+     * etcetera.
+     *
+     * @var resource
+     */
+    private $ch;
+
     public function __construct()
     {
+        $this->ch = curl_init();
+    }
+
+    public function __destruct()
+    {
+        curl_close($this->ch);
     }
 
     /**
-     * @param $url
-     * @param $options
+     * @param string $url
+     * @param array  $options
      *
-     * @return CurlRequest
+     * @return Request
      */
     public function request($url, $options)
     {
+        if (!empty($options['curlHandle'])) {
+            $options['curlHandle'] = $this->ch;
```

Comment:
Why set this only if empty? And why override it if it's passed in?

Reply:
Gah, thanks. I originally was using `isset`.